tensorboard/webapp/widgets/line_chart_v2/line_chart_internal_utils.ts

```diff
@@ -22,44 +23,55 @@ import {isWebGl2Supported} from './lib/utils';
 /**
  * Returns extent, min and max values of each dimensions, of all data series points.
  *
+ * When ignoreYOutliers is true, it will filter out values that are approximately [1] less
+ * than 5th and greater than 95th percentile when calculating y-extent.
+ *
  * Note that it excludes auxillary data points and invisible data series.
  *
- * TODO(stephanwlee): add support for ignoreOutlier.
+ * [1]: Uses R-7 method for approximation: https://github.com/d3/d3-array#quantile
  */
 export function computeDataSeriesExtent(
   data: DataSeries[],
-  metadataMap: DataSeriesMetadataMap
+  metadataMap: DataSeriesMetadataMap,
+  ignoreYOutliers: boolean
 ): {x: [number, number] | undefined; y: [number, number] | undefined} {
-  let xMin = Infinity;
-  let xMax = -Infinity;
-  let yMin = Infinity;
-  let yMax = -Infinity;
-
-  let xExtentChanged = false;
-  let yExtentChanged = false;
+  let yPoints: number[] = [];
+  const xPoints: number[] = [];
 
+  let pointIndex = 0;
   for (const {id, points} of data) {
     const meta = metadataMap[id];
     if (!meta || meta.aux || !meta.visible) continue;
 
     for (let index = 0; index < points.length; index++) {
       const {x, y} = points[index];
-      if (!Number.isNaN(x)) {
-        xMin = Math.min(xMin, x);
-        xMax = Math.max(xMax, x);
-        xExtentChanged = true;
+      if (Number.isFinite(x)) {
+        xPoints.push(x);
       }
-      if (!Number.isNaN(y)) {
-        yMin = Math.min(yMin, y);
-        yMax = Math.max(yMax, y);
-        yExtentChanged = true;
+      if (Number.isFinite(y)) {
+        yPoints.push(y);
       }
+      pointIndex++;
     }
   }
 
+  yPoints.sort(d3.ascending);
+
+  if (ignoreYOutliers && yPoints.length > 2) {
+    const aY = d3.quantile(yPoints, 0.05)!;
```

Comment:
Quantile sorted is a recent addition that we cannot use since we are stuck on d3 5.7.0 (internal version)

Reply:
Ah, got it.  In that case though I think that's a stronger case to just reimplement the index-calculation logic (which is just one line) and do a direct lookup into the sorted array rather than doing two unnecessary scans.